src/docs/guides/data-protection/README.md

```diff
@@ -0,0 +1,161 @@
+
+If not supplied, the `BlobName` will automatically default to a folder per tenant configuration: `Sites/tenant_name/DataProtectionKeys.xml`
+
+#### Configuring a Container per Tenant
+
+For more granular isolation, you can configure a separate container for each tenant:
+
+```json
+{
+  "OrchardCore": {
+    "OrchardCore_DataProtection_Azure": {
+      "ConnectionString": "DefaultEndpointsProtocol=https;AccountName=<myaccountname>;AccountKey=<myaccountkey>;EndpointSuffix=core.windows.net",
+      "ContainerName": "{{ ShellSettings.Name }}-dataprotection",
+      "BlobName": "{{ ShellSettings.Name }}DataProtectionKeys.xml",
+      "CreateContainer": true
+    }
+  }
+}
+```
+
+!!! note
+    When templating the `ContainerName` using `{{ ShellSettings.Name }}`, the tenant's name will be automatically lowercased. However, you must also ensure the `ContainerName` conforms to other Azure Blob naming conventions as outlined in Azure's documentation.
+
+!!! important
+    Only the default Liquid filters and tags are available during parsing of the Liquid template. Extra filters like `slugify` will not be available.
+
+## Redis Data Protection
+
+The `OrchardCore.Redis.DataProtection` feature enables storing data protection keys in Redis, providing a high-performance solution for distributed applications.
+
+### Purpose
+
+Redis Data Protection enables sharing data protection keys across multiple application instances using Redis as the centralized storage. This is ideal for load-balanced environments where low-latency key access is required.
+
+### Prerequisites
+
+Before configuring Redis Data Protection, ensure you have:
+
+1. A running Redis instance
+2. The `OrchardCore.Redis` module enabled
+3. The `OrchardCore.Redis.DataProtection` feature enabled
+
+### Configuration
+
+First, configure the basic Redis connection in your `appsettings.json`:
+
+```json
+{
+  "OrchardCore": {
+    "OrchardCore_Redis": {
+      "Configuration": "<your-redis-connection-string>",
+      "InstancePrefix": "MyApp:",
+      "AllowAdmin": true
+    }
+  }
+}
+```
+
+#### Configuration Options
+
+- **Configuration**: The Redis connection string (required)
+- **InstancePrefix**: A prefix to be added to all Redis keys (optional)
+- **AllowAdmin**: Whether to allow admin commands (optional, defaults to false)
+
+### Key Storage Structure
+
+Redis Data Protection stores keys using the following pattern:
+
+```
+{InstancePrefix}{TenantName}:DataProtection-Keys
+```
+
+For example, with an InstancePrefix of "MyApp:" and a tenant named "Default", the key would be:
+
+```
+MyApp:Default:DataProtection-Keys
+```
+
+### Persistence Considerations
+
+!!! warning
+    Redis data protection is enabled. Ensure your Redis server has a backup strategy in place to prevent data loss. Use either AOF (Append-Only File) or RDB (Redis Database) persistence.
```

Comment:
```suggestion
    Data protection keyrings are not cache files and must be kept in durable storage. Ensure that your Redis server has a backup strategy in place to prevent data loss. Use either AOF (Append-Only File) or RDB (Redis Database) persistence.
```